### ISSUE TITLE
Verification of TTL for transactions seem to be missing

### ISSUE THREAD

**Issue by gleber-da** [digital-asset/daml-on-x-example#18]
There seem to be no calls into `TimeModeChecker.checkTtl` anywhere from this code. I was not able to find it transitively called as well. This might mean that the TTLs for proposed transactions are not properly checked.

**Comment by dajmaki** [comment]
Yep this should be added. I'm not sure it's such a great idea to have two methods to check these time parameters as (as one can see) it is easy to miss one of them. The effect of not having this check in was that the maximum record time provided in the request was not checked against the time model, which in effect could have caused problems if an application re-submits a command that was erroneously accepted. Of course command deduplication should then kick in...
